src/java.base/share/classes/java/util/stream/Gatherers.java

```diff
@@ -430,16 +428,12 @@ boolean flush(long atLeastN, Downstream<? super R> downstream) {
                                 try {
                                     next.thread.join();
                                 } catch (InterruptedException ie) {
-                                    interrupted = true; // ignore, for now, and restore later
+                                    // We ignore interrupts here because we cannot proceed
+                                    // until all spawned threads have exited
```

Comment:
Why discard instead of keeping existing deferral? Swallowing interrupt requests generally breaks cancellation.

Reply:
@dholmes-ora Good question. It is to avoid the situation where re-interruption and/or timing of interrupt leads to difference in behavior. But your question led me to re-examing an assumption so I might have to reconsider the best path forward. In the meantime I'll close the PR.